docker-compose files and yarn dependency cleanup

c0m3d1an opened the pull request:
Changes to be committed:
	new file:   Dockerfile
	new file:   docker-compose.yml
	modified:   package.json
	modified:   yarn.lock

dcaslin commented on package.json:
```diff
@@ -41,15 +42,18 @@
     "core-js": "^2.6.1",
     "hammerjs": "^2.0.8",
     "idb-keyval": "^3.2.0",
+    "jquery": "^1.9.1",
```
What's with the package.json changes besides the docker start?

dcaslin commented:
Dockerfile stuff all looks great! Can you either: 

1) Restore the package.json and yarn.lock back to match the origin

2) I can just grab this Dockerfile and paste the docker run in package.json into the latest

c0m3d1an commented:
My assumption is that you started this project using npm and switched (at some point) to yarn.  When I tried to build your site using yarn I got multiple errors for packages that would not build because their dependencies were not installed.  I updated the yarn and package.json files to include those dependencies so ng serve would spin up correctly.

Additionally, since Bungo REQUIRES the site to be SSL (the return URL MUST start with https://) I needed a line in there for a custom start command with the --ssl switch(which creates dummy certs if they don't exist).  I don't believe the jquery include nor the correction of the line with the tab at the end is required, but the line with "docker-start": "ng serve --host 0.0.0.0 --ssl", would be EXTREMELY helpful to maintain.

As for yarn.lock, your program WILL NOT BUILD on any recent linux I tried building it on due to version and dependency requirements, so the yarn.lock changes (were)are absolutely required for this program to build successfully.  Due to the age of angular you're using and the age of the build dependencies, if the versions don't exist in the yarn.lock, it tries to pull the most recent versions and fails building the components that depend on them.

I see your commit and conflict request only require me pulling edits from package.json, and you'll paste them back in there.  If that IS the only requirement, then I can re-sub with the Dockerfile, docker-compose.yml, and yarn.lock.

Make sense?  LMK.

dcaslin commented:
I'm so sorry, I somehow missed your response! I'll read this in more detail tomorrow and get back. In the meantime: 

- You are correct that I use yarn as my package manager (I build on Mac and Windows 10 for personal dev)

- You are correct about HTTPS. That -ssl flag is really nice! What I do now is just hack it for local dev where I run my webpack server on http and have to manually delete the http**s** from the URL on the Bungie sign in call.

dcaslin commented:
> I see your commit and conflict request only require me pulling edits from package.json, and you'll paste them back in there. If that IS the only requirement, then I can re-sub with the Dockerfile, docker-compose.yml, and yarn.lock.

This would be perfect! Please go ahead with that, thanks!

c0m3d1an commented:
I did actually re-submit the branch as discussed after my comments above.  :-)  It still works without the deps in the package.json, although it does compile with dependency errors.  (Although still completely fails without the deps in the yarn.lock).

dcaslin commented:
I figured out the jquery mystery, bootstrap (which I'm using for pure CSS and no JS) had it as a dependency. Just merged, thanks!

c0m3d1an commented:
Oh awesome, good to hear!  I am NOT remotely a node or angular expert, so I was pretty much just flailing blindly in an attempt to get the docker code working.  :-)

dcaslin commented:
Np! I just pushed 7.8.0 after adding back in the docker-start command. If you get a chance give it a run and see if it actually works (hopefully I didn't break anything)

c0m3d1an commented:
Works perfectly!  Yay!
